In [ ]:
print ("hello word")

hello word


In [5]:
# Mount google drive
from google.colab import drive
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [13]:
# Import libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [9]:
data = pd.read_csv('/content/gdrive/MyDrive/DATA SET/n_movies.csv')

**1. Missing Value Check and Handling**

In [15]:
data.info () 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9957 entries, 0 to 9956
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   title        9957 non-null   object 
 1   year         9430 non-null   object 
 2   certificate  6504 non-null   object 
 3   duration     7921 non-null   object 
 4   genre        9884 non-null   object 
 5   rating       8784 non-null   float64
 6   description  9957 non-null   object 
 7   stars        9957 non-null   object 
 8   votes        8784 non-null   object 
dtypes: float64(1), object(8)
memory usage: 700.2+ KB


Raw Data have 9 columns and 9957 row
all the data except title, description, stars have "missing values"

In [16]:
# number of missing values per column
data.isna().sum()

title             0
year            527
certificate    3453
duration       2036
genre            73
rating         1173
description       0
stars             0
votes          1173
dtype: int64

In [18]:
# drop missing values
data = data.dropna()

In [19]:
# check missing values
data.isna().sum()

title          0
year           0
certificate    0
duration       0
genre          0
rating         0
description    0
stars          0
votes          0
dtype: int64

Data already to use, Done! There's no more missing values

**2. Duplicate check simple**

In [21]:
# number of duplicated data
data.duplicated().sum()

0

In [22]:
data.head ()

title         year certificate duration  \
0               Cobra Kai     (2018– )       TV-14   30 min   
1               The Crown     (2016– )       TV-MA   58 min   
2        Better Call Saul  (2015–2022)       TV-MA   46 min   
3           Devil in Ohio       (2022)       TV-MA  356 min   
4  Cyberpunk: Edgerunners     (2022– )       TV-MA   24 min   

                          genre  rating  \
0         Action, Comedy, Drama     8.5   
1     Biography, Drama, History     8.7   
2                  Crime, Drama     8.9   
3        Drama, Horror, Mystery     5.9   
4  Animation, Action, Adventure     8.6   

                                         description  \
0  Decades after their 1984 All Valley Karate Tou...   
1  Follows the political rivalries and romance of...   
2  The trials and tribulations of criminal lawyer...   
3  When a psychiatrist shelters a mysterious cult...   
4  A Street Kid trying to survive in a technology...   

                                               stars    votes  
0  ['Ralph Macchio, ', 'William Zabka, ', 'Courtn...  177,031  
1  ['Claire Foy, ', 'Olivia Colman, ', 'Imelda St...  199,885  
2  ['Bob Odenkirk, ', 'Rhea Seehorn, ', 'Jonathan...  501,384  
3  ['Emily Deschanel, ', 'Sam Jaeger, ', 'Gerardo...    9,773  
4  ['Zach Aguilar, ', 'Kenichiro Ohashi, ', 'Emi ...   15,413